Assets/Scripts/Framework/Puzzles/SequencePuzzle/ColorSequenceManager.cs

```diff
@@ -0,0 +1,120 @@
+using UnityEngine;
+using System;
+using UnityEngine.Events;
+
+
+public sealed class ColorSequenceManager : MonoBehaviour, SequenceManager<EnumColor>
+{
+    public EnumColor[] Sequence { get; set; }  = new EnumColor[5];
+    public EnumColor[] SubmittedSequence { get; set; } = new EnumColor[5];
+
+    [SerializeField] private UnityEvent onSolved = new UnityEvent();
+    [SerializeField] private UnityEvent onReset = new UnityEvent();
+
+    private int sequenceLength = 3;
```

Comment:
```suggestion
    [SerializeField] private int sequenceLength;
```
Beter om instelbaar te maken via inspector